src/measure_theory/function/uniform_integrable.lean

```diff
@@ -834,6 +834,68 @@ begin
   end
 end
 
+lemma uniform_integrable.spec (hp : p ≠ 0) (hp' : p ≠ ∞)
+  (hfu : uniform_integrable f p μ) {ε : ℝ} (hε : 0 < ε) :
+  ∃ C : ℝ≥0, ∀ i, snorm ({x | C ≤ ∥f i x∥₊}.indicator (f i)) p μ ≤ ennreal.of_real ε :=
+begin
+  obtain ⟨hf₀, hfu, M, hM⟩ := hfu,
+  by_cases hMzero : M = 0,
```

Comment:
Not sure you need to specialize `M = 0` here: if your assumption holds for `M`, it holds for `max M 1`, so you can use `max M 1` instead of `M` in the proof below and everything goes through.